src/ripple/app/ledger/impl/LedgerConsensusImp.cpp

```diff
@@ -1016,6 +1022,12 @@ void LedgerConsensusImp::accept (std::shared_ptr<SHAMap> set)
 
         buildLCL->updateSkipList ();
 
+        if (buildLCL->info().seq > v2_ledger_seq_switch &&
+            !buildLCL->stateMap().is_v2())
```

Comment:
You may be correct, but I am not 100% positive.  `SHAMap::fetchNodeFromDB` can switch back to v1 if a mismatch is detected.  It is possible that this may happen between line 988 and here.  This is such tricky code that it needs both belt and suspenders everywhere.